### PR TITLE
Update all user timeslots to :preferred if all timeslots are :available

### PR DESCRIPTION
**Pull request by pizzani**

### CODE REVIEW

File: src/pairing_scheduler/core.clj [hunk 1]
```diff
@@ -107,13 +107,26 @@
                         [guest-id (individual-score guest-id context)]))
                  (into {}))))
 
+;; if a user has no preferred timeslots,
+;; update all available timeslots to preferred.
```
Comment (rafd) [inline src/pairing_scheduler/core.clj:111]: Make this comment a docstring

File: src/pairing_scheduler/core.clj [hunk 2]
```diff
@@ -107,13 +107,26 @@
...
+(defn update-preferred
+  [{:keys [schedule availabilities] :as context}]
+  (->> availabilities
+       (map (fn [[key value]]
+          (if (= 0 (count (filter (fn [x] (= (x 2) :preferred)) value)))
+            [key (set (map (fn [x] [(x 0) (x 1) :preferred]) value))]
+            [key value])))
+       (into {})
+       (assoc context :availabilities)))
```
Comment (rafd) [inline src/pairing_scheduler/core.clj:120]: Instead, do `(update context :availabilities (fn [availabilities]  .... ))`

File: src/pairing_scheduler/core.clj [hunk 3]
```diff
@@ -107,13 +107,26 @@
                         [guest-id (individual-score guest-id context)]))
                  (into {}))))
 
+;; if a user has no preferred timeslots,
+;; update all available timeslots to preferred.
+(defn update-preferred
+  [{:keys [schedule availabilities] :as context}]
+  (->> availabilities
+       (map (fn [[key value]]
```
Comment (rafd) [inline src/pairing_scheduler/core.clj:115]: Use better names than `key` and `value`

File: src/pairing_scheduler/core.clj [hunk 4]
```diff
@@ -107,13 +107,26 @@
...
+          (if (= 0 (count (filter (fn [x] (= (x 2) :preferred)) value)))
```
Comment (rafd) [inline src/pairing_scheduler/core.clj:116]: instead of `(x 2)` .   `(fn [[_ _ preference]]  ...)`

Comment (rafd) [reply src/pairing_scheduler/core.clj:116]: instead of count, filter and = 0 , use `some`

File: src/pairing_scheduler/core.clj [hunk 5]
```diff
@@ -107,13 +107,26 @@
...
+            [key (set (map (fn [x] [(x 0) (x 1) :preferred]) value))]
```
Comment (rafd) [inline src/pairing_scheduler/core.clj:117]: Maybe use `->> `

File: src/pairing_scheduler/core.clj (hunk 2 above)
Comment (pizzani) [reply src/pairing_scheduler/core.clj:163]: couldn't get (update context :availabilities (fn [availabilities] .... )) to work.

**Comment by rafd** [comment]
Thanks. Refactored slightly and moved the function to the import namespace. Merged by squash, so I'm closing this PR manually.

**Pull request closed without merging**